### ISSUE TITLE
Silence banner of GAP.jl when loaded from another package

### ISSUE THREAD

**Issue by albinahlback** [oscar-system/GAP.jl#765]
I am developing a package which uses GAP.jl. I would like to silence the banner of GAP from within this package, and I do not know how to do that. For example, writing in the REPL
```
julia> ENV["GAP_PRINT_BANNER"] = "false"
"false"

julia> using MyPackage
```
works as intended. However, I cannot seem to do the same when writing it inside the package.

How can I silence the banner from within my package?

**Comment by fingolfin** [comment]
OK, I see: the problem is that `ENV["GAP_PRINT_BANNER"] = "false"` in the main body of your packages source code only has an effect during precompilation, so it won't affect the `using`. And if you place it into `__init__` it is also useless, because the `__init__` of `GAP` of course is processed first.


Anyway, ideally you shouldn't even *have* to do anything. We'd like to print those banners only if you type `using GAP` on the REPL, not if you do it in your own package. We have a crude hack in place to ensure that if `using GAP` is done from GAP, it doesn't show the banner. But so far none of us figured out how to generically distinguish the two scenarios.

So, either someone does (and then I'll happily add it here, and in the other OSCAR packages); or else we need another workaround, but I have no idea how that could work... :-/.

**Comment by fingolfin** [comment]
Actually, I take that back: I just discovered we have https://github.com/Nemocas/Nemo.jl/pull/1055 (but this was only added to Nemo and Hecke, not to GAP.jl/Singular.jl/Polymake.jl ...

I will look closer at this and see what I can do to fix this in 0.7.4 / 0.8.0 (the 0.7.3 release I just made is only there to fix compatibility with Julia nightlies)

**Comment by fingolfin** [comment]
@albinahlback Does PR #768 resolve the issue for you?
